aws_lambda_powertools/utilities/streaming/_s3_seekable_io.py

```diff
@@ -67,6 +68,7 @@ def __init__(
         self._sdk_options = sdk_options
         self._sdk_options["Bucket"] = bucket
         self._sdk_options["Key"] = key
+        self._user_agent_added = False
```

Comment:
```suggestion
        self._has_user_agent = False
```

prefer a more explicit name.

```python
if self._has_user_agent:
```

vs

```python
if self._user_agent_added:
```